assets/js/challenges.js

```diff
@@ -61,9 +61,27 @@ Alpine.data("Challenge", () => ({
   solves: [],
   response: null,
 
+  init() {
+    const self = this;
+    const submit = document.querySelector("#challenge-submit");
+
+    // eslint-disable-next-line no-undef
+    if (challenge && challenge.submit) {
+      submit.addEventListener(
+        "click",
+        Alpine.debounce(this.submitCustomChallenge.bind(self), 500)
+      );
+    } else {
+      submit.addEventListener(
+        "click",
+        Alpine.debounce(this.submitChallenge.bind(self), 500)
+      );
+    }
```

Comment:
Why do we need this and the submitCustomChallenge function?

Reply:
I think I understand now but I think this strategy is unnecessary and we should focus on submitChallenge being the main function for submitting challenges.